code/crates/sync/src/handle.rs

```diff
@@ -412,23 +412,31 @@ where
         .sync_height
         .increment_by(state.config.parallel_requests);
 
-    loop {
-        // Check if already have a pending request or validation for this height.
-        if !state.has_pending_value_request(&height) && !state.has_pending_value_validation(&height)
-        {
-            let Some(peer) = state.random_peer_with_tip_at_or_above(height) else {
-                debug!(height.sync = %height, "No peer to request sync from");
-                // No peer reached this height yet, we can stop here.
-                break;
-            };
-
-            request_value_from_peer(&co, state, metrics, height, peer).await?;
-        } else {
-            debug!(height.sync = %height, "Already have a pending request or validation for this height");
+    // Find out the first height for which we do not have a pending request or validation.
+    while state.has_pending_value_request(&height) || state.has_pending_value_validation(&height) {
+        height = height.increment();
+        if height >= limit {
```

Comment:
If we reach this case (i.e., all heights in the current range have a pending request or validation), the `height` variable is equal to the `limit`, which is not in the range. But the following code will request a value for this height, as the limit is checked at the end of the `loop`.